### PR TITLE
pd_client: add backoff for the reconnect retries

### PR DESCRIPTION
**Pull request by nolouch**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/15428 
ref https://github.com/tikv/pd/issues/6556

What's Changed:

```commit-message
 pc_client: add store-level backoff for the reconnect retries
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

**In 50 TiKVs, 1M regions cluster: use `pd-failpoint-binary`, and ingest the failpoint.**

1. enable fail point, like gRPC is throttling, cannot read from etcd.
`curl -X PUT -d 'return(10)' http://tc-pd-1.tc-pd-peer.csn-simulator-big-cluster-vd62g.svc:2379/pd/api/v1/fail/github.com/tikv/pd/pkg/etcdutil/SlowEtcdKVGet`

2. simulate pd lost leader
`curl -X PUT -d 'return("2346857576170797299")' http://tc-pd-1.tc-pd-peer.csn-simulator-big-cluster-vd62g.svc:2379/pd/api/v1/fail/github.com/tikv/pd/server/exitCampaignLeader`

#### Before
image

#### After
image

### Release note 

```release-note
 add backoff for the reconnect retries in pd-client
```

### CODE REVIEW

**Comment by nolouch** [comment]
PTAL @rleungx  @Connor1996

File: components/pd_client/src/util.rs [hunk 1]
```diff
@@ -900,6 +901,33 @@ impl PdConnector {
...
+        }
+    }
+    pub fn next_backoff(&mut self) -> Duration {
+        self.interval = std::cmp::min(self.interval * 2, MAX_BACKOFF);
+        self.interval
+    }
+
+    pub fn get_interval(&self) -> Duration {
+        self.interval
+    }
```
Comment (rleungx) [inline components/pd_client/src/util.rs:924]: How about:
```suggestion
    pub fn get_interval(&self) -> Duration {
        self.interval = std::cmp::min(self.interval * 2, MAX_BACKOFF);
        self.interval
    }
```

Comment (nolouch) [reply components/pd_client/src/util.rs:924]: We need this two interfaces in the current implement.

**Review by rleungx — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**